pandas/core/construction.py

```diff
@@ -509,15 +509,20 @@ def ensure_wrapped_if_datetimelike(arr):
 
 def sanitize_masked_array(data: ma.MaskedArray) -> np.ndarray:
     """
-    Convert numpy MaskedArray to ensure mask is softened.
+    Convert numpy MaskedArray to ensure mask is softened,
+
     """
     mask = ma.getmaskarray(data)
     if mask.any():
-        dtype, fill_value = maybe_promote(data.dtype, np.nan)
-        dtype = cast(np.dtype, dtype)
-        data = ma.asarray(data.astype(dtype, copy=True))
-        data.soften_mask()  # set hardmask False if it was True
-        data[mask] = fill_value
+        dtype = cast(np.dtype, data.dtype)
+        if isinstance(dtype, ExtensionDtype) and dtype.name.startswith("Masked"):
```

Comment:
I haven't checked thoroughly, but I think you want `dtype._can_hold_na` instead of having logic depend on the name of the dtype.